File: core/Core/Models/Transaction.cs
using System;
using System.Collections.Generic;

namespace Core.Models;

public partial class Transaction
{
    public int TransactionId { get; set; }

    public required int SourceAccount { get; set; }

    public required int TargetAccount { get; set; }

    public required decimal Amount { get; set; }

    public required string Currency { get; set; }

    public required DateTime CreatedAt { get; set; }

    public required string Status { get; set; }

    public required Account SourceAccountNavigation { get; set; }

    public required Account TargetAccountNavigation { get; set; }

    public virtual ICollection<TransactionFee> TransactionFees { get; set; } = new List<TransactionFee>();

    public virtual ICollection<TransactionCategory> Categories { get; set; } = new List<TransactionCategory>();
}

File: analytic/domain/transaction/transaction.py
from datetime import datetime
from decimal import Decimal


class Transaction:
    def __init__(self,
                 transaction_id: int,
                 source_account: int | None,
                 target_account: int | None,
                 amount: Decimal,
                 currency: str,
                 created_at: datetime | None,
                 status: str | None):
        self.transaction_id = transaction_id
        self.source_account = source_account
        self.target_account = target_account
        self.amount = amount
        self.currency = currency
        self.created_at = created_at
        self.status = status
